src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -1737,7 +1737,14 @@ class PartitionClauseSegment(BaseSegment):
         "PARTITION",
         "BY",
         Delimited(
-            Ref("ColumnReferenceSegment"),
+            OneOf(
+                Ref("ColumnReferenceSegment"),
+                Bracketed(
+                    Ref("SelectStatementSegment"),
+                ),
+                Ref("FunctionSegment"),
+                Ref("VariableIdentifierSegment"),
+            ),
```

Comment:
Since this whole section is within the `Delimited` section just want to check if something like this is possible?
```sql
sum(foo) OVER (PARTITION BY (SELECT foo from bar), (SELECT foo from bar), foo, sum(foo))
```

Reply:
@jpy-git I'm not sure about the aggregate function within the PARTITION BY clause, but otherwise yes.  I just ran this nonsense on my personal database successfully:

```
SELECT 
	ROW_NUMBER() OVER 
		(PARTITION BY 
			(SELECT ms.MediaTypeName FROM dbo.MediaTypes ms WHERE ms.MediaTypeID = f.mediatypeid),
			(SELECT ms.MediaTypeName FROM dbo.MediaTypes ms WHERE ms.MediaTypeID = f.mediatypeid),
			mediatypeid,
			COALESCE(mediatypeid, filmid)
		ORDER BY AdjustedPriorityScore DESC),
	*
FROM films f

```

Reply:
Awesome thanks for checking! this is good to merge then 👍